main.c

```diff
@@ -1,3 +1,48 @@
+#include <stdint.h>
+
+#include "am1808.h"
+
+// Shared RAM (control input)
+typedef struct shared_ram {
+    // Declare this as a u32 to force more-efficient codegen
+    uint32_t pwms;
+} shared_ram;
+// XXX: The real address in use here is 0x80010000
+// There appears to be a compiler bug where ctable entries with the MSB set
+// do not get optimized correctly. We lie to the PRU compiler here, but the
+// hardware indeed contains the correct address we *actually* want.
+static volatile shared_ram * const SHARED = (volatile shared_ram *)0x7f010000;
+#pragma ctable_entry 30 0x7f010000
+
+// LED definitions PRU1_R30[n]
+#define LED0    12
+#define LED1    10
+#define LED2    13
+#define LED3    11
+
+static inline void update_pwm(uint8_t val, uint8_t time_now, uint32_t gpio_bit) {
```

Comment:
```suggestion
static inline void update_pwm(uint8_t val, uint8_t time_now, uint8_t gpio_bit) {
```

saves a few instructions